src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,132 @@
+package duke.utils;
+
+import duke.DukeException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * The `Parser` class is responsible for parsing user input and extracting relevant information
+ * for further processing by the Duke chatbot.
+ */
+public class Parser {
+    private List<String> inputArray;
+
+    /**
+     * Constructs a new `Parser` with an empty input array.
+     */
+    public Parser() {
+
+    }
+
+    /**
+     * Constructs a new `Parser` with user input and splits it into an array of words.
+     * @param userInput The user input string to be parsed.
+     */
+    public Parser(String userInput) {
+        this.inputArray = Arrays.asList(userInput.split(" "));
+    }
+
+    /**
+     * Extracts the command keyword from the user input.
+     * @return The extracted command keyword in uppercase.
+     */
+    public String inputCommand() {
```

Comment:
Would the code here be clearer if a verb was used to name the method instead? One possibility could be `getInputCommand`.